vllm/v1/core/block_pool.py

```diff
@@ -223,8 +227,15 @@ def cache_full_blocks(
             return
         new_full_blocks = blocks[num_cached_blocks:num_full_blocks]
         assert len(request.block_hashes) >= num_full_blocks
-        new_block_hashes = request.block_hashes[num_cached_blocks:]
+        if block_size == self.hash_block_size:
+            block_hashes: BlockHashList = request.block_hashes
+        else:
+            assert block_size % self.hash_block_size == 0
+            block_hashes = BlockHashListWithBlockSize(
```

Comment:
```suggestion
            # Get the block_hash with block_size tokens per block from reqeust.block_hashes with hash_block_size tokens per block
            block_hashes = BlockHashListWithBlockSize(
```
Add a comment like this? You can also try to make it more clear. I feel it's hard for first-time reader to get what this piece of code is doing now.